Lib/pathlib.py

```diff
@@ -873,6 +873,10 @@ def absolute(self):
             cwd = self._flavour.abspath(self.drive)
         else:
             cwd = os.getcwd()
+            if not self.root and not self._tail:
+                path = self.with_segments(cwd)
+                path._str = cwd  # Fully normalized string from getcwd().
+                return path
```

Comment:
Optional nit (feel free to ignore): there's a lot of paths in this module ;) Maybe `new_path`, `new`, or even just `ret` would be a better name for this variable?